superset-frontend/packages/superset-ui-core/src/time-comparison/getTimeOffset.ts

```diff
@@ -78,6 +78,10 @@ export const parseDttmToDate = (
         now.setMonth(now.getMonth() - 1);
       }
       return now;
+    case 'previous quarter':
```

Comment:
Could you add test cases for this calculation? For example, the month of May should be quarter number 2.

Reply:
I can add tests, but the purpose here isn't to return a quarter number. It's to add the option for "previous calendar quarter" to a date filter (and obviously calculate that correctly). Currently the "custom" options include previous week, month and year but not quarter.
![image](https://github.com/user-attachments/assets/627f0ba3-b7ac-4684-a4f3-807da4b48147)

Reply:
Done :)